clients/vault/src/horizon/client.rs

```diff
@@ -0,0 +1,279 @@
+use crate::{
+	error::Error,
+	horizon::{HorizonTransactionsResponse, PagingToken, Transaction},
+	oracle::ScpMessageHandler,
+};
+use async_trait::async_trait;
+use service::Error as ServiceError;
+use sp_std::{convert::From, str, vec::Vec};
+use std::{convert::TryInto, sync::Arc, time::Duration};
+
+use stellar_relay_lib::sdk::{Hash, SecretKey};
+use tokio::{sync::Mutex, time::sleep};
+
+const POLL_INTERVAL: u64 = 5000;
+
+pub const fn horizon_url(is_public_network: bool) -> &'static str {
+	if is_public_network {
+		"https://horizon.stellar.org"
+	} else {
+		"https://horizon-testnet.stellar.org"
+	}
+}
+
+// todo: temporary struct, as the real IssueRequest definition is in another branch.
+#[derive(Clone, PartialEq)]
+pub struct IssueRequest {
+	pub issue_id: Hash,
+}
+
+impl From<Vec<u8>> for IssueRequest {
+	fn from(val: Vec<u8>) -> Self {
+		IssueRequest { issue_id: val[0..31].try_into().unwrap() }
+	}
+}
+
+pub type IssueRequests = Vec<IssueRequest>;
+
+#[async_trait]
+pub trait HorizonClient {
+	async fn get_transactions(&self, url: &str) -> Result<HorizonTransactionsResponse, Error>;
+}
+
+#[async_trait]
+impl HorizonClient for reqwest::Client {
+	async fn get_transactions(&self, url: &str) -> Result<HorizonTransactionsResponse, Error> {
+		self.get(url)
+			.send()
+			.await
+			.map_err(|_| Error::HttpFetchingError)?
+			.json::<HorizonTransactionsResponse>()
+			.await
+			.map_err(|_| Error::HttpFetchingError)
+	}
+}
+
+struct HorizonFetcher<C: HorizonClient> {
+	client: C,
+	vault_secret_key: String,
+	last_tx_id: Option<Vec<u8>>,
+}
+
+impl<C: HorizonClient> HorizonFetcher<C> {
+	pub fn new(client: C, vault_secret_key: String) -> Self {
+		Self { client, vault_secret_key, last_tx_id: None }
+	}
+
+	/// Fetch recent transactions from remote and deserialize to HorizonResponse
+	/// Since the limit in the request url is set to one it will always fetch just one
+	async fn fetch_latest_txs(
+		&self,
+		is_public_network: bool,
+		next_page: PagingToken,
+	) -> Result<HorizonTransactionsResponse, Error> {
+		let vault_keypair: SecretKey = SecretKey::from_encoding(&self.vault_secret_key).unwrap();
+		let vault_address = vault_keypair.get_public();
+
+		let mut request_url = String::from(horizon_url(is_public_network)) +
+			"/accounts/" + str::from_utf8(vault_address.to_encoding().as_slice())
+			.map_err(|_| Error::HttpFetchingError)? +
+			"/transactions?";
+
+		if !next_page.is_empty() {
+			request_url.push_str(&format!("cursor={}&", next_page));
+		}
+		request_url += "order=desc&limit=20";
```

Comment:
I guess this should actually be `order=asc&limit=20`, or does it work? Because I think otherwise you get the transactions older than that paging token but we want the transactions that are more recent. 

Nit: It might also be a good idea to extract the `limit` to a constant and parse it into the string.